### PR TITLE
Simplify tuist init

### PR DESCRIPTION
**Pull request by fortmarek**
### Short description 📝

This PR simplifies the project generated by `tuist init` command. Additionally, we're removing the UIKit and AppKit templates in favor of only having a SwiftUI project.

### How to test the changes locally 🧐

Run `tuist init`, also for other platforms. Not `watchos` and `visionos` currently don't work and should be handled in a separate PR (see more [here](https://github.com/tuist/tuist/issues/5607))

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint:fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistKit/Commands/InitCommand.swift [hunk 1]
```diff
@@ -37,7 +37,7 @@ public struct InitCommand: ParsableCommand, HasTrackableParameters {
 
     @Option(
         name: .shortAndLong,
-        help: "The name of the project. If it's not passed (Default: Name of the directory)"
+        help: "The name of the project. If it's not passed (Default: Name of the current directory)"
```
Comment (danyf90) [inline Sources/TuistKit/Commands/InitCommand.swift:40]:
```suggestion
        help: "The name of the project (Default: Name of the current directory)"
```

File: Templates/AppProject.stencil [hunk 2]
```diff
@@ -1,29 +1,31 @@
...
+			destinations: .{{ platform }},
+			product: .unitTests,
+			bundleId: "io.tuist.{{ name }}Tests",
+			infoPlist: .default,
+			sources: ["{{ name }}/Tests/**"],
+			resources: [],
+			dependencies: [.target(name: "{{ name }}")]
+		),
+	]
+)
```
Comment (danyf90) [inline Templates/AppProject.stencil:31]: nit: maybe it's just me, but I would indent using 2 spaces

File: Tests/TuistAcceptanceTests/InitAcceptanceTests.swift [hunk 3]
```diff
@@ -27,20 +27,6 @@ final class InitAcceptanceTestiOSApp: TuistAcceptanceTestCase {
...
-final class InitAcceptanceTestSwiftUImacOSApp: TuistAcceptanceTestCase {
-    func test_init_swift_ui_macos_app() async throws {
-        try run(InitCommand.self, "--platform", "macos", "--name", "MyApp", "--template", "swiftui")
-        try await run(BuildCommand.self)
-    }
-}
-
 // TODO: Fix
 // final class InitAcceptanceTestSwiftUtvOSApp: TuistAcceptanceTestCase {
 //    func test_init_swift_ui_tvos_app() async throws {
```
Comment (danyf90) [inline Tests/TuistAcceptanceTests/InitAcceptanceTests.swift:32]: This can be removed as well, right?

**Review by danyf90 — APPROVED** [review]
One one side, I agree that we should have a simple init project to avoid being too opinionated and reduce the maintenance.
On the other, I wonder whether such a streamlined project would bring any benefit to the user.

I am maybe more inclined in removing the `tuist init` command altogether, and having some DocC tutorial about creating a basic app with tuist 🤔

**Comment by fortmarek** [comment]
> I am maybe more inclined in removing the tuist init command altogether, and having some DocC tutorial about creating a basic app with tuist 🤔

As discussed on Slack, I disagree with this – I strongly believe there's value in having a simple project generated by `tuist init` with some basic stuff like unit tests, launch screen, gitignore, etc. Just so that you can actually immediately generate a project, run the app, and iterate from there. The project generated by `tuist init` is similar to the `New project` created by Xcode.

I will address the rest of the feedback 🙏

File: Templates/AppProject.stencil (hunk 2 above)
Comment (fortmarek) [reply Templates/AppProject.stencil:31]: Indentation in tuist is 4 spaces, also for all of our fixtures, so I'd keep that

Comment (danyf90) [reply Templates/AppProject.stencil:31]: Good to me to use 4 spaces, but this is not 4 spaces, it's tabs :D
Also, I think some lines are misaligned as they use spaces instead

Comment (fortmarek) [reply Templates/AppProject.stencil:31]: Yeah, that was there before 🙈 let me update it

File: Templates/default/AppProject.stencil [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        .target(
+            name: "{{ name }}",
+            destinations: .{{ platform }},
+            product: .app,
+			bundleId: "io.tuist.{{ name }}",
+			infoPlist: .extendingDefault(
+                with: [
+                    "UILaunchStoryboardName": "LaunchScreen.storyboard",
+                ]
+            ),
```
Comment (danyf90) [inline Templates/default/AppProject.stencil:15]: Almost :D 
```suggestion
            product: .app,
            bundleId: "io.tuist.{{ name }}",
            infoPlist: .extendingDefault(with: [
                "UILaunchStoryboardName": "LaunchScreen.storyboard",
            ]),
```

**Review by danyf90 — APPROVED** [review]

**Pull request merged**